File: sticker/src/main/kotlin/chatox/sticker/model/StickerPack.kt
package chatox.sticker.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.time.ZonedDateTime

@Document
data class StickerPack<PreviewMetadataType>(
        @Id
        val id: String,

        @Indexed
        val name: String,

        @Indexed
        val author: String?,
        val description: String,

        @Indexed
        val createdBy: String,
        val createdAt: ZonedDateTime,
        val updatedAt: ZonedDateTime? = null,
        val preview: Upload<PreviewMetadataType>
)

File: client/src/api/types/response/StickerPack.ts
import {Upload} from "./Upload";
import {GifUploadMetadata} from "./GifUploadMetadata";
import {ImageUploadMetadata} from "./ImageUploadMetadata";
import {Sticker} from "./Sticker";

export interface StickerPack {
    author?: string,
    createdAt: string,
    description: string,
    id: string,
    name: string,
    preview: Upload<GifUploadMetadata | ImageUploadMetadata>,
    stickers: Sticker[]
}
